planner/optimize.go

```diff
@@ -105,7 +105,7 @@ func optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is in
 
 	// Handle the logical plan statement, use cascades planner if enabled.
 	if sctx.GetSessionVars().EnableCascadesPlanner {
-		return cascades.FindBestPlan(sctx, logic)
+		return cascades.NewOptimizer().FindBestPlan(sctx, logic)
```

Comment:
Can we create a singleton for it, for example `defaultOptimizer`? Then we can use it globally, don't need to create a new `Optimizer` object for each query.

Reply:
Sure, I created it in `planner/cascades/optimizer`.